File: LeetCode/1523-count-odd-numbers-in-an-interval-range/1523-count-odd-numbers-in-an-interval-range.js
/**
 * @param {number} low
 * @param {number} high
 * @return {number}
 */
var countOdds = function(low, high) {
    let count = 0;
    
    if (low % 2 === 0 && high % 2 === 0) {
        count = (high - low) / 2;
        
        return count;
    }
    
    count = Math.floor((high - low) / 2) + 1;
    
    return count;
};

File: 1523-count-odd-numbers-in-an-interval-range/1523-count-odd-numbers-in-an-interval-range.java
class Solution {
    public int countOdds(int low, int high) {
        //low high가 홀수 인지.
        // 3 4 5 6 7  -> 7과 3 사이 3개 숫자. 하나가 홀수면 -1 두개다 홀수면 -2 
        // 2 3 4 5 6 7 8 9 둘다 짝수면 high - low - 1 /2 , 하나면 high - low -1 / 2 + 1 
        // 3 4 5 6 7 8 9 둘다 홀수 floor 5 / 2 + 2  
        
        if (low % 2 == 0 && high % 2 == 0) {
            return (int)Math.ceil((high - (double)low - 1) / 2); 
        }
        
        if (low % 2 == 0 ^ high % 2 == 0) {
            return (high - low - 1) / 2 + 1;
        }
        
        if (low == high) {
            return 1;
        }
        
        return (int)Math.floor((high - low - 1) / 2) + 2;
    }
}